Library/Homebrew/cmd/log.rb

```diff
@@ -14,14 +14,24 @@ def log
 
   private
 
-  def git_log(path=nil)
+  def git_log(path = nil)
     if File.exist? "#{`git rev-parse --show-toplevel`.chomp}/.git/shallow"
       opoo <<-EOS.undent
         The git repository is a shallow clone therefore the filtering may be incorrect.
         Use `git fetch --unshallow` to get the full repository.
       EOS
     end
+
+    if ARGV.size > 2 then raise <<-EOS.undent
+      Usage is:
+        brew log formula
+      or:
+        brew log formula version
+    EOS
+    end
+
     args = ARGV.options_only
+    args += ["--grep", ARGV[1]] unless ARGV[1].nil?
```

Comment:
Not you specifically. Apologies if my reply erred on the side of being too blunt.